### PR TITLE
[DUOS-2145][risk=low] Add GET dataset by id V2 endpoint

### PR DESCRIPTION
**Pull request by connorlbark**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2145

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/assets/paths/datasetByIdV2.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  description: Finds Dataset and DatasetProperties of the Dataset specified by id.
+  parameters:
+    - name: id
+      in: path
+      description: ID of the Dataset
+      required: true
+      schema:
+        type: integer
+  tags:
+    - Datasets
```
Comment (rushtong) [inline src/main/resources/assets/paths/datasetByIdV2.yaml]: I updated the tag to the singular version for consistency in https://github.com/DataBiosphere/consent/pull/1761
```suggestion
  tags:
    - Dataset
```

File: src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java [hunk 2]
```diff
@@ -209,6 +209,7 @@ public Response findAllDatasetsAvailableToUser(@Auth AuthUser authUser) {
     }
 
     @GET
+    @Deprecated
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java:254]: We should mark this in the api docs as deprecated as well.

**Review by rushtong — COMMENTED** [review]
Two minor comments, then 👍🏽

**Review by JVThomas — APPROVED** [review]
Code looks good. Dataset is coming in upon request

**Review by rushtong — APPROVED** [review]
👍🏽

**Pull request merged**